subsys/net/ip/net_shell.c

```diff
@@ -2933,6 +2934,9 @@ static int ping_ipv6(const struct shell *shell,
 #if defined(CONFIG_NET_ROUTE)
 	struct net_route_entry *route;
 #endif
+	if (iface_idx != -1) {
+		iface = net_if_get_by_index(iface_idx);
+	}
```

Comment:
This can be removed.

And then below, we can do things like this. So if the iface is not set, we figure it out from the destination address and if everything fails, only then use the defaut interface
```
if (!iface) {
    iface = net_if_ipv6_select_src_iface(...);
    if (!iface) {
    ...
         if (nbr) {
          ...
         } else {
            iface = net_if_get_default();
         }
    }
}
```